dockerfiles/init/manifests/codenvy.env

```diff
@@ -86,17 +86,121 @@
 #CODENVY_SERVER_XMX=2048
 
 
-
 ########################################################################################
 #####                                                                              #####
-#####                                   DOCKER                                     #####
+#####                                 NETWORKING                                   #####
 #####                                                                              #####
-# IP Address
-#     The IP address of the Docker daemon that is running on your host. We do a 
-#     self-discovery to set this. You can combine this with values of DOCKER_HOST to 
-#     change communications from socket to TCP.
+#
+# Affects how the Codenvy server connects to the Docker Daemon, how the Che server and
+# workspaces establish connections to one another, and also how remote browser clients
+# should discover and connect to workspaces.
+#
+# Codenvy goes through a progression algorithm to establish the protocol, IP address and
+# port to establish communications with it is booting or starting a workspace.
+#
+# Browser --> Codenvy Server
+#    1. Default is 'http://localhost:${SERVER_PORT}/wsmaster/api'.
+#    2. Else use the value of CODENVY_API
+#
+# Codenvy Server --> Docker Daemon Progression:
+#    1. Use the value of CODENVY_DOCKER_DAEMON__URL
+#    2. Else, use the value of DOCKER_HOST system variable
+#    3. Else, use Unix socket over unix:///var/run/docker.sock
+#
+# Codenvy Server --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
```

Comment:
For terminal and ws-agent next properties influence connection machine.proxy_wsagent_server_url_template
machine.proxy_terminal_server_url_template

Strategy should not be changed since we have only 1 strategy designed to work in codenvy.
Also there is some host dependent logic influenced by some variables, but it is better to ask Florent about that.